AzureMonitor/Scripts/Create-AzDiagPolicy.ps1

```diff
@@ -1776,12 +1803,28 @@ function New-PolicyInitiative
     $PolicyRSIDs = $PolicyRSIDs.substring(0,$PolicyRSIDs.length -3)
     $PolicyDefParams = $PolicyDefParams.substring(0,$PolicyDefParams.length -3)
     
+    # Adding support for Management Group deployment scope.  If parameter switch is used for -ManagementGroupDeployment, we'll put the right JSON in to support
+    if($ManagementGroupDeployment)
+    {
+        $MGJSONParam = @'
+{
+    "TargetMGID": {
+        "type": "string",
+        "defaultValue": ""
+    }
+}
+'@
+    }
+    else
+    {
+        $MGJSONParam = '{}'
+    }
     # Build Template reference for Policy Initiative
     $InitiativeTemplate = @'
 {
     "$schema": "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
```

Comment:
you were quicker than me with PR, so I'll put my insights in comments

- schema should be adjusted for mg/ sub deployment

I've tried this way:

``` powershell
function New-PolicyInitiative
(
    [Parameter(Mandatory=$True)]
    [string]$PolicyBag,
    [Parameter(Mandatory=$True)]
    [string]$PolicyRSIDs,
    [Parameter(Mandatory=$True)]
    [string]$PolicyDefParams,
    [Parameter(Mandatory=$True)]
    [string]$Parameters,
    [Parameter(Mandatory=$True)]
    [string]$sinkDest,
    [Parameter(Mandatory=$True)]
    [string]$InitiativeDisplayName,
    [Parameter(Mandatory=$True)]
    [string]$InitiativeName,
    [Parameter(Mandatory=$True)]
    [ValidateSet('Subscription','ManagementGroup')]
    [string]$Scope
)
{
    # Scrub trailing commas
    $PolicyRSIDs = $PolicyRSIDs.substring(0,$PolicyRSIDs.length -3)
    $PolicyDefParams = $PolicyDefParams.substring(0,$PolicyDefParams.length -3)
    
    # Build Template reference for Policy Initiative
    $InitiativeTemplate = @'
{
    "$schema": "<AZURE DIAG INITIATIVE SCHEMA>",
    "contentVersion": "1.0.0.0",
    "parameters": <AZURE DIAG INITIATIVE PARAMETERS>,
    "resources": [
        <AzurePolicyPropertyBag>
        {
            "type": "Microsoft.Authorization/policySetDefinitions",
            "apiVersion": "2019-09-01",
            "name": "<AZURE DIAG INITIATIVE NAME>",
            "dependsOn": [
<Policy INIT RESIDs>
            ],
            "properties": {
                "displayName": "<AZURE DIAG INITIATIVE DISPLAY NAME>",
                "description": "This initiative configures application Azure resources to forward diagnostic logs and metrics to an Azure Diagnostics sink point.",
                "metadata": {
                    "category": "Monitoring"
                },
                "parameters": <ParametersGoHere>,
                "policyDefinitions": [
                    <PolicyDefParams>
                ]
            }
        }
    ]
}
'@

    # TBD - schema
    $ScopeToArmTemplateSchemaMapping = @{
        Subscription = 'https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#'
        ManagementGroup = 'https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#'
    }
    $ArmTemplateSchema = $ScopeToArmTemplateSchemaMapping[$Scope]

    $InitiativeParameters = $null
    if ($Scope -eq 'Subscription') {
        $InitiativeParameters = '{}'
    }
    elseif ($Scope -eq 'ManagementGroup')
    {
        $InitiativeParameters = @'
{
    "scope": {
        "type": "string",
        "metadata": {
            "description": "TBD"
        }
    }
}
'@
    }

    # Update Policy Initiative reference strings according to what we've discovered
    $InitiativeTemplate = $InitiativeTemplate.Replace("<AZURE DIAG INITIATIVE NAME>", $InitiativeName)
    $InitiativeTemplate = $InitiativeTemplate.Replace("<AZURE DIAG INITIATIVE DISPLAY NAME>", $InitiativeDisplayName)
    $InitiativeTemplate = $InitiativeTemplate.Replace("<AZURE DIAG INITIATIVE SCHEMA>", $ArmTemplateSchema )
    $InitiativeTemplate = $InitiativeTemplate.Replace("<AZURE DIAG INITIATIVE PARAMETERS>", $InitiativeParameters )
    $InitiativeTemplate = $InitiativeTemplate.Replace("<AzurePolicyPropertyBag>", $PolicyBag)
    $InitiativeTemplate = $InitiativeTemplate.Replace("<Policy INIT RESIDs>", $PolicyRSIDs)
    $InitiativeTemplate = $InitiativeTemplate.Replace("<ParametersGoHere>", $Parameters)
    $InitiativeTemplate = $InitiativeTemplate.Replace("<PolicyDefParams>", $PolicyDefParams)

    $InitiativeTemplate
}
```

Reply:
Of course then also  New-PolicyInitiative invoke must be adjusted:

```powershell
        $scope = 'Subscription'
        if($ExportInitiativeMgScope)
        {
            $scope = 'ManagementGroup'
        }

        # Building the Policy Initiative (Note only one sink point per policy initiative [Log Analytics or EventHub])
        $PolicyInititiative = New-PolicyInitiative -PolicyBag $PolicyBag -PolicyRSIDs $PolicyRSIDs -PolicyDefParams $PolicyDefParams -Parameters $PolicyJSON[0] -sinkDest $sinkDest -InitiativeDisplayName $InitiativeDisplayName -InitiativeName $InitiativeName -Scope $scope
        
```

$ExportInitiativeMgScope was my name for the switch. I see you use 'ManagementGroupDeployment'.

Reply:
@kwiecek Are you ok with the existing updated PR that already has updated schema?  I was traveling when you made this recommendation so making sure you see latest and can make recommendations there.  I made updates to this PR that should support schema switch for Initiative as well as ensuring the Management Group option is supported if requested during ARM template generation..